Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -285,6 +387,50 @@ - (void)updateAvatarView
 
 #pragma mark - AvatarViewHelperDelegate
 
++ (BOOL)shouldDisplayProfileViewOnLaunch
+{
+    // Only nag until the user sets a profile _name_.  Profile names are
+    // recommended; profile avatars are optional.
+    if ([OWSProfileManager sharedManager].localProfileName.length > 0) {
+        return NO;
+    }
+
+    NSTimeInterval kProfileNagFrequency = kDayInterval * 30;
+    NSDate *_Nullable lastPresentedDate =
+        [[[TSStorageManager sharedManager] dbReadConnection] dateForKey:kProfileView_LastPresentedDate
```

Comment:
Oh, I see, you're saying that we should make a new db connection that hangs on the class for consistency.  

I'd rather not keep the connection dangling around forever when we'll use it very rarely, especially since we read/write from this particular state very rarely and a race condition is pretty unlikely. 

Hmm, point taken though, I guess I'll just use the storage manager's shared read/write connection.